integration-tests/src/test/java/oracle/weblogic/kubernetes/ItHPA.java

```diff
@@ -0,0 +1,235 @@
+      "https://github.com/kubernetes-sigs/metrics-server/releases/download/metrics-server-helm-chart-3.8.2/components.yaml";
+
+  /**
+   * Assigns unique namespaces for operator and domains.
+   * Pull WebLogic image if running tests in Kind cluster.
+   * Installs operator.
+   *
+   * @param namespaces injected by JUnit
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(2) List<String> namespaces) {
+    logger = getLogger();
+    logger.info("Assign a unique namespace for operator");
+    assertNotNull(namespaces.get(0), "Namespace is null");
+    opNamespace = namespaces.get(0);
+
+    logger.info("Assign a unique namespace for WebLogic domain");
+    assertNotNull(namespaces.get(1), "Namespace is null");
+    domainNamespace = namespaces.get(1);
+
+    // set the service account name for the operator
+    opServiceAccount = opNamespace + "-sa";
+
+    // install and verify operator with REST API
+    installAndVerifyOperator(opNamespace, opServiceAccount, true, 0, domainNamespace);
+
+    // create pull secrets for WebLogic image when running in non Kind Kubernetes cluster
+    // this secret is used only for non-kind cluster
+    createBaseRepoSecret(domainNamespace);
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    adminSecretName = "weblogic-credentials";
+    createSecretWithUsernamePassword(adminSecretName, domainNamespace,
+        ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT);
+
+    // create encryption secret
+    logger.info("Create encryption secret");
+    encryptionSecretName = "encryptionsecret";
+    createSecretWithUsernamePassword(encryptionSecretName, domainNamespace,
+        "weblogicenc", "weblogicenc");
+
+    domain = createDomainResource(
+        domainUid,
+        domainNamespace,
+        MII_BASIC_IMAGE_NAME + ":" + MII_BASIC_IMAGE_TAG,
+        adminSecretName,
+        new String[]{TEST_IMAGES_REPO_SECRET_NAME},
+        encryptionSecretName
+    );
+
+    // create cluster resouce with limits and requests in serverPod
+    ClusterResource clusterResource =
+        createClusterResource(clusterResName, wlClusterName, domainNamespace, replicaCount);
+    clusterResource.getSpec()
+        .serverPod(new ServerPod().resources(
+            new V1ResourceRequirements()
+                .putLimitsItem("cpu", Quantity.fromString("2"))
+                .putLimitsItem("memory", Quantity.fromString("2Gi"))
+                .putRequestsItem("cpu", Quantity.fromString("250m"))
+                .putRequestsItem("memory", Quantity.fromString("768Mi"))));
+    logger.info("Creating cluster {0} in namespace {1}", clusterResName, domainNamespace);
+    createClusterAndVerify(clusterResource);
+
+    // set cluster references
+    domain.getSpec().withCluster(new V1LocalObjectReference().name(clusterResName));
+
+    // verify the domain custom resource is created
+    createDomainAndVerify(domain, domainNamespace);
+  }
+
+  @BeforeEach
```

Comment:
We can move these steps to BeforeAll()

Reply:
kept it in beforeAll so that if we add another test method its better to check BeforeEach, there is one more use case we are discussing for autoscaling ...if we are not going to add that, I can move it to beforeAll

Reply:
leaving it at beforeEach as I will be working on another use case